cocos/game/director.ts

```diff
@@ -819,6 +819,7 @@ export declare namespace Director {
 
     export type OnUnload = () => void;
 
+    export type OnScenePreloaded = (error?: Error) => void;
```

Comment:
Yep, rename it to OnScenePreloaded is more clear. But it will break compatibility.

Reply:
Yeah the main problem is that the current interface is really misleading since the extra argument. I wanted instead to at least make it valid

Reply:
Sorry, as keep compatible is important. So can not merge this PR. And as v3.8.5 is released. So this branch will be deleted, only add a tag for 3.8.5.